subxt/src/backend/legacy/mod.rs

```diff
@@ -84,183 +85,249 @@ impl<T: Config + Send + Sync + 'static> Backend<T> for LegacyBackend<T> {
         keys: Vec<Vec<u8>>,
         at: T::Hash,
     ) -> Result<StreamOfResults<StorageResponse>, Error> {
-        let methods = self.methods.clone();
+        async fn fut<T: Config>(
+            keys: Vec<Vec<u8>>,
+            at: T::Hash,
+            methods: LegacyRpcMethods<T>,
+        ) -> Result<StreamOfResults<StorageResponse>, Error> {
```

Comment:
Nit: do we need this separate fut function when all of the ones below are inline in the `retry`? :)

Reply:
I had some issues to get it to compile IIRC but lemme check again.

Reply:
Yeah I wondered if that was the case :)